### DISCUSSION TITLE
Relative position of agent after executing the action.

### DISCUSSION THREAD

**Discussion by aseembehl** [eleurent/highway-env#176] [General]
Is it possible to somehow compute the relative position of the agent after taking a step in the highway environment?

**Comment by eleurent** [comment]
The simplest way is to access the agent position in an abolute frame directly, and then perform any computation that you require. For instance:
```
p, h = env.vehicle.position, env.vehicle.heading
env.step(action)
next_p, next_h = env.vehicle.position, env.vehicle.heading
print(next_p - p, next_h - h)
```

The agent position and heading can also be obtained as an observation (as in `obs, reward, info, done = env.step(action)`) if you use the `KinematicsObservation` in the environment configuration with `absolute=True`. Note that using `absolute=False` describes the other vehicles relatively to the agent, which I don't think is what you need.
